modules/widgets/twitter-timeline.php

```diff
@@ -188,11 +169,42 @@ public function widget( $args, $instance ) {
 		 */
 		$timeline_placeholder = apply_filters( 'jetpack_twitter_timeline_placeholder', $timeline_placeholder );
 
-		echo esc_html( $timeline_placeholder ) . '</a>';
+		$type      = ( isset( $instance['type'] ) ? $instance['type'] : '' );
+		$widget_id = ( isset( $instance['widget-id'] ) ? $instance['widget-id'] : '' );
+
+		if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+			$width   = ! empty( $instance['width'] ) ? $instance['width'] : 600;
+			$height  = ! empty( $instance['height'] ) ? $instance['height'] : 480;
+			$output .= '<amp-twitter' . $data_attrs . ' layout="responsive" data-timeline-source-type="profile" data-timeline-screen-name="' . esc_attr( $widget_id ) . '" width="' . absint( $width ) . '" height="' . absint( $height ) . '">'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			$output .= esc_html( $timeline_placeholder ) . '</amp-twitter>';
+
+			echo $output . $args['after_widget']; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			return;
+		}
```

Comment:
Note that this is currently needed because the AMP plugin has a bug in how it recognizes markup for Twitter embeds. Namely, at the moment it is only recognizing `div` elements with the `twitter-tweet` class name:

https://github.com/ampproject/amp-wp/blob/4880f0f58daaf07685854be8574ff25d76ff583e/includes/embeds/class-amp-twitter-embed-handler.php#L146

It should be expanded to also recognize `twitter-timeline` as well. If so, then I believe this condition can be removed.

Reply:
I've filed an issue to fix this in the AMP plugin: https://github.com/ampproject/amp-wp/issues/4653

With that done, it may make sense to remove this condition as it won't strictly be needed anymore. That being said, it is not bad to output `<amp-twitter>` markup from the start! It can be somewhat preferable to relying on the AMP plugin do the conversion.